### DISCUSSION TITLE
LEDs go green on black screen

### DISCUSSION THREAD

**Discussion by jkirkcaldy** [awawa-dev/HyperHDR#92] [Q&A]
Using a raspberry pi with ws281x led strips with a USB hdmi grabber. 

On a black screen my LEDs go green. I can confirm that the screen is actually black as it's an oled TV and the screen itself is off when the LEDs are showing green. 

If I use the android hyperion hdr grabber there is no issue.

**Comment by awawa-dev** [comment]
Do you have HDR tone mapping enabled?
Please take and upload a screenshot of that black screen from the HyperHDR video stream preview.

**Reply by jkirkcaldy** [reply]
Here you go:

<img width="1436" alt="Screenshot 2021-07-30 at 20 03 38" src="https://user-images.githubusercontent.com/26098592/127700451-00ba0cab-6b68-4804-b971-fdc08add8855.png">

<img width="537" alt="Screenshot 2021-07-30 at 20 03 45" src="https://user-images.githubusercontent.com/26098592/127700335-c6b70908-f2c3-4079-a08a-c7c1c48ab202.png">

**Reply by awawa-dev** [reply]
Probably clicking on the 'screenshot' button could provide more accurate results but you can test it on your own.
The sample you provided is near black ````#070707```` and perfectly balanced: each of RGB component are equal so greenish light on your led comes from incorrect calibration in the processing tab or it's a led strip characteristic (personally I found that only RGBW sk6812 provides very natural color/grey level in low light). But if the android grabbers works fine then check your processing tab because it seems that video capturing is fine.

**Comment by S0und** [comment]
Are you sure that this just a "incorrect calibration in the processing tab or it's a led strip characteristic"?

So i'm running Hyperion over a year, with a Aliexpress adalight (ch340 which is an Arduino Uno clone), driven by [hyperion-android-grabber](https://github.com/abrenoch/hyperion-android-grabber), and i never had this issue.

This week i switched over to a hdmi capture card (ZasLuke Game Capture Card) and to HyperHDR. In my case balanced blacks are red. 

here is how a 0,0,0 black bmp looks like with stock Hyperion alpha 10 settings, hdmi capture:

https://i.imgur.com/TOYGmqi.png

same image on HyperHDR 17beta

https://i.imgur.com/jcuQ2fc.png

I have to note that, obviously this is not a realistic exposure level, it looks much brighter in the photo, but it's noticeable. 

Again, i'm using stock settings on both service, the only thing i've changed is to setup USB capture and the LEDs

```
HyperHDR Server: 
- Build:           (HEAD detached at 9881cf8) (Awawa-84a0fdc/9881cf8-1629326009)
- Build time:      Aug 19 2021 19:02:57
- Git Remote:      https://github.com/awawa-dev/HyperHDR
- Version:         17.0.0.0beta
- UI Lang:         auto (BrowserLang: en-US)
- UI Access:       expert
- Avail Capt:      Linux (V4L2)
- Database:        read/write

HyperHDR Server OS: 
- Distribution:   Raspbian GNU/Linux 10 (buster)
- Architecture:   arm
- CPU Model:      ARMv7 Processor rev 3 (v7l)
- CPU Type:       Raspberry Pi 4 Model B Rev 1.1
- CPU Revision:   b03111
- CPU Hardware:   BCM2711
- Kernel:         linux (5.10.60-v7l+ (WS: 32))
- Qt Version:     5.11.3
- Browser:        Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:91.0) Gecko/20100101 Firefox/91.0 
```

**Comment by awawa-dev** [comment]
Nothing indicates anything else so far. There is no magic ..the only things that could alter captured color data (if in fact the grabber provides perfect black) is:
- custom settings in particular LED driver like in Philips hue (but that's not the case), let's assume that the Arduino sketch renders RGB 1:1
- HDR tone mapping (but that's not the case either on default settings)
- custom LUT tables for YUV codecs (even without HDR tone mapping, probably not the case either)
- image to LED mapping algorithm type (but for zeroes result should be similar, processing tab) and always active the processing tab color calibration and brightness manipulation.  

So in the typical case it leaves only processing tab...  
  
If you want to dig deeper, 3 things are necessary:
1 backup of your full v17 configuration from general tab
2 screen shot for grabber 'black' from the led preview ( not screenshot from OS like last time). Browser MJPEG inner stream could introduce some noise but it's not important now.
3 change led driver to debug:'file' type and provide the result content from the output file for the same 'black' video stream

**Reply by S0und** [reply]
Turning off Image Processsing > Classic HyperHDR calibration, solved my issue, (0,0,0) black turns off the LEDs

Thanks!

**Reply by awawa-dev** [reply]
Glad it works but the classic calibration is neutral for processing for gamma = 1 (non default but  neutral, it's a sanity check):
![obraz](https://user-images.githubusercontent.com/69086569/132140538-cc65c4b7-c295-4d14-a76d-780c7b52dde4.png)
Hardcoded (16,16,16) on input => (16,16,16) on LED output after classic calibration.

For defaults HyperHDR settings (gamma is 1.5, not neutral):
![obraz](https://user-images.githubusercontent.com/69086569/132140598-a294add7-46a4-4eaf-81f6-4dfc5b240e95.png)
Still it's gray not reddish. Hardcoded (16,16,16) =>(4,4,4) on LED output after classic calibration with gammas (1.5).

**Reply by S0und** [reply]
Just to get this right, if the input pixel color is (0,0,0) you're still output (4,4,4) as a minimum? The reason why am asking this, because i have an other CH340 (which i soft-bricked in the past) and installed [Adalight-FastLED](https://github.com/dmadison/Adalight-FastLED) on it. With `Classic HyperHDR calibration` ON, it outputing a greyish color, i'm assuming it's (4,4,4), instead of the previous redish color.

**Reply by awawa-dev** [reply]
I think I asked you to provide the config, image test sample and output file. For default settings for black the output is (1,1,1) because of default minimal backlit in the processing tab: grey not reddish again. If you disable minimal backlit then you'll get (0,0,0)

**Comment by pjmpessers** [comment]
For me it helped to set grabber's video format from MJPEG to YUYV.
